scalding-core/src/main/scala/com/twitter/scalding/reducer_estimation/Common.scala

```diff
@@ -181,6 +187,13 @@ object ReducerEstimatorStepStrategy extends FlowStepStrategy[JobConf] {
       // if still None, make it '-1' to make it simpler to log
       val numReducers = combinedEstimator.estimateReducers(info)
 
+      numReducers.foreach { n =>
+        val configuredMax = conf.getInt(EstimatorConfig.maxEstimatedReducersKey, EstimatorConfig.defaultMaxEstimatedReducers)
+        if (n > configuredMax) {
+          throw new IllegalArgumentException(s"Reducer estimator estimated $n reducers, which is more than the configured maximum of $configuredMax")
```

Comment:
I too prefer not to fail jobs. As for max, I suggest theoretical max mappers / 5. We could set  it 10000 for clusters with 50K-split jobs. Anyways I am not worried too much of the default in the code here because I expect we'll configure it explicitly for important pipelines.

The layering of estimator in Scalding is not as ideal. If we deferred to mapreduce JobSubmitter we could run it after getSplits and actually know the number of mappers. But this discussion is beyond the scope of this issue.